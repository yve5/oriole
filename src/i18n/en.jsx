const I18N_DICTIONARY = {
  LANG: 'en',
  LANGUAGE: 'English',
  DATE_FORMAT: 'MM/dd/yyyy',
  LOCALE: 'enUS',
  'About us': 'About us',
  'An error has been detected. Please try again.':
    'An error has been detected. Please try again.',
  Buttons: 'Buttons',
  Changelog: 'Changelog',
  Collapse: 'Collapse',
  Expand: 'Expand',
  Internationalization: 'Internationalization',
  Paging: 'Paging',
  'The request has been sent successfully.':
    'The request has been sent successfully.',
};

export default I18N_DICTIONARY;
