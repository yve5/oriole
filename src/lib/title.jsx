import Title from '../shared/title';

export { Title };
