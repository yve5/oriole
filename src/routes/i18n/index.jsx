import I18nPage from './components/I18n';

export default I18nPage;
