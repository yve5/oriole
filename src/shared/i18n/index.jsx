import LangSwitch from './components/LangSwitch';
import I18n from './components/I18n';
import i18nReducer from './reducer';

export { I18n, LangSwitch, i18nReducer };
