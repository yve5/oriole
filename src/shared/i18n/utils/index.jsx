import { checkTermOrder, convertCollectionToArray } from './Term';
import { getInitialState } from './InitialState';
import { checkMissingTerms } from './Missing';
import { formatDate } from './Format';

export {
  formatDate,
  checkTermOrder,
  getInitialState,
  checkMissingTerms,
  convertCollectionToArray,
};
