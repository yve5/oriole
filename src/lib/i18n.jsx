import { I18n, LangSwitch, i18nReducer } from '../shared/i18n';

export { I18n, LangSwitch, i18nReducer };
