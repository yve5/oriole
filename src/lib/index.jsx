import { I18n, LangSelect, LangSwitch, i18nReducer } from '../shared/i18n';
import {
  trimTerms,
  formatDate,
  checkTermOrder,
  checkMissingTerms,
  convertCollection,
  checkLocalTermOrder,
  convertLocalCollection,
} from '../shared/i18n/utils';

import SvgContainer, { getIconComponent } from '../shared/icons';

import Theme from '../shared/theme';
import Title from '../shared/theme';

export {
  I18n,
  Theme,
  Title,
  LangSelect,
  LangSwitch,
  SvgContainer,
  trimTerms,
  formatDate,
  i18nReducer,
  checkTermOrder,
  getIconComponent,
  checkMissingTerms,
  convertCollection,
  checkLocalTermOrder,
  convertLocalCollection,
};
