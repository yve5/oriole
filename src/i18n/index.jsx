import fr from './fr';
import en from './en';
import { tsl } from '../shared/i18n/utils/Translate';

const i18n = {
  tsl: (term, args) => tsl(fr, term, args),
  dictionaries: [fr, en],
  lang: fr.LANG,
};

export default i18n;
