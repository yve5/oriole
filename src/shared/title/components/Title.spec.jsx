import React from 'react';
import configureMockStore from 'redux-mock-store';

import { thunk } from 'redux-thunk';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { act, fireEvent, render } from '@testing-library/react';

import i18n from '../../../i18n';
import { Title } from '../../../lib/title';

const mockStore = configureMockStore([thunk]);

const fakeUrl = '/url/example';
const fakeTitle =
  'Lorem ipsum dolor sit amet consectetur adipisicing elit. Facilis suscipit ad modi exercitationem laudantium iure, perspiciatis a quam aliquam assumenda pariatur ratione laborum repellat sapiente rem voluptates quidem eligendi minus?';

const defaultStore = { i18n };

describe('Title', () => {
  it('should match expected snapshot. default', () => {
    const { asFragment } = render(
      <Provider store={mockStore(defaultStore)}>
        <MemoryRouter>
          <Title />
        </MemoryRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match expected snapshot. backTo url', () => {
    const { asFragment } = render(
      <Provider store={mockStore(defaultStore)}>
        <MemoryRouter>
          <Title backTo={fakeUrl} />
        </MemoryRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  const theButton = (
    <button className="btn btn-link btn-sm" type="button">
      Export
    </button>
  );

  it('should match expected snapshot. optional', () => {
    const { asFragment } = render(
      <Provider store={mockStore(defaultStore)}>
        <MemoryRouter>
          <Title optional={theButton} />
        </MemoryRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match expected snapshot. undefined title', () => {
    const { asFragment } = render(
      <Provider store={mockStore(defaultStore)}>
        <MemoryRouter>
          <Title title={undefined} />
        </MemoryRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match expected snapshot. null title', () => {
    const { asFragment } = render(
      <Provider store={mockStore(defaultStore)}>
        <MemoryRouter>
          <Title title={null} />
        </MemoryRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match expected snapshot. empty title', () => {
    const { asFragment } = render(
      <Provider store={mockStore(defaultStore)}>
        <MemoryRouter>
          <Title title="" />
        </MemoryRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match expected snapshot. plain title', () => {
    const { asFragment } = render(
      <Provider store={mockStore(defaultStore)}>
        <MemoryRouter>
          <Title title="Hello World" />
        </MemoryRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match expected snapshot. long title', async () => {
    const { asFragment, getByText } = render(
      <Provider store={mockStore(defaultStore)}>
        <MemoryRouter>
          <Title title={fakeTitle} />
        </MemoryRouter>
      </Provider>
    );

    await act(async () => {
      fireEvent.click(
        getByText(
          'Lorem ipsum dolor sit amet consectetur adipisicing elit. Facilis suscipit ad modi exercitationem laudantium iure, perspiciatis a quam aliquam assumenda pariatur ratione laborum repellat sapiente rem voluptates quidem ...'
        )
      );
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it('should match expected snapshot. i18n term', () => {
    const { asFragment } = render(
      <Provider store={mockStore(defaultStore)}>
        <MemoryRouter>
          <Title i18n="Hello World" />
        </MemoryRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match expected snapshot. i18n term and title', () => {
    const { asFragment } = render(
      <Provider store={mockStore(defaultStore)}>
        <MemoryRouter>
          <Title i18n="Hello World" title="New account" />
        </MemoryRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match expected snapshot. skeleton', () => {
    const { asFragment } = render(
      <Provider store={mockStore(defaultStore)}>
        <MemoryRouter>
          <Title
            title={
              <span className="Skeleton Skeleton--title" title="Loading..." />
            }
          />
        </MemoryRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match expected snapshot. no print', () => {
    const { asFragment } = render(
      <Provider store={mockStore(defaultStore)}>
        <MemoryRouter>
          <Title backTo={fakeUrl} title="Hello World" hideInPrint />
        </MemoryRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
