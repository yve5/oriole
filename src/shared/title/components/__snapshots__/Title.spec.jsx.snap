// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Title > should match expected snapshot. backTo url 1`] = `
<DocumentFragment>
  <a
    class="title__btn"
    href="/url/example"
  >
    <svg
      class="title__icon"
      fill="currentColor"
      height="16"
      viewBox="0 96 960 960"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M561 816 320 575l241-241 43 43-198 198 198 198-43 43Z"
        stroke="currentColor"
        stroke-width="12"
      />
    </svg>
    Retour
  </a>
  <h1
    class="title"
  >
    ...
  </h1>
</DocumentFragment>
`;

exports[`Title > should match expected snapshot. default 1`] = `
<DocumentFragment>
  <h1
    class="title title--alone"
  >
    ...
  </h1>
</DocumentFragment>
`;

exports[`Title > should match expected snapshot. empty title 1`] = `
<DocumentFragment>
  <h1
    class="title title--alone"
  >
    ...
  </h1>
</DocumentFragment>
`;

exports[`Title > should match expected snapshot. i18n term 1`] = `
<DocumentFragment>
  <h1
    class="title title--alone"
  >
    Bonjour le monde
  </h1>
</DocumentFragment>
`;

exports[`Title > should match expected snapshot. i18n term and title 1`] = `
<DocumentFragment>
  <h1
    class="title title--alone"
  >
    New account
  </h1>
</DocumentFragment>
`;

exports[`Title > should match expected snapshot. long title 1`] = `
<DocumentFragment>
  <h1
    class="title title--alone title--long"
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Facilis suscipit ad modi exercitationem laudantium iure, perspiciatis a quam aliquam assumenda pariatur ratione laborum repellat sapiente rem voluptates quidem eligendi minus?
  </h1>
</DocumentFragment>
`;

exports[`Title > should match expected snapshot. no print 1`] = `
<DocumentFragment>
  <a
    class="title__btn title--no-print"
    href="/url/example"
  >
    <svg
      class="title__icon"
      fill="currentColor"
      height="16"
      viewBox="0 96 960 960"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M561 816 320 575l241-241 43 43-198 198 198 198-43 43Z"
        stroke="currentColor"
        stroke-width="12"
      />
    </svg>
    Retour
  </a>
  <h1
    class="title title--no-print"
  >
    Hello World
  </h1>
</DocumentFragment>
`;

exports[`Title > should match expected snapshot. null title 1`] = `
<DocumentFragment>
  <h1
    class="title title--alone"
  >
    ...
  </h1>
</DocumentFragment>
`;

exports[`Title > should match expected snapshot. optional 1`] = `
<DocumentFragment>
  <button
    class="btn btn-link btn-sm"
    type="button"
  >
    Export
  </button>
  <h1
    class="title title--alone"
  >
    ...
  </h1>
</DocumentFragment>
`;

exports[`Title > should match expected snapshot. plain title 1`] = `
<DocumentFragment>
  <h1
    class="title title--alone"
  >
    Hello World
  </h1>
</DocumentFragment>
`;

exports[`Title > should match expected snapshot. skeleton 1`] = `
<DocumentFragment>
  <h1
    class="title title--alone"
  >
    <span
      class="Skeleton Skeleton--title"
      title="Loading..."
    />
  </h1>
</DocumentFragment>
`;

exports[`Title > should match expected snapshot. undefined title 1`] = `
<DocumentFragment>
  <h1
    class="title title--alone"
  >
    ...
  </h1>
</DocumentFragment>
`;
